astroid/bases.py

```diff
@@ -328,6 +328,12 @@ def getitem(self, index, context=None):
             raise exceptions.InferenceError(
                 "Could not find __getitem__ for {node!r}.", node=self, context=context
             )
+        if len(method.args.arguments) != 2:  # (self, index)
+            raise exceptions.AstroidTypeError(
+                "__getitem__ for {node!r} does not have correct signature",
+                node=self,
+                context=context,
+            )
```

Comment:
This fixes the `InvalidGetitem` test in in `test_invalid_subscripts`

Reply:
I'm trying to match the style of a few other inference functions: when it is known that code would raise a `Value/Index/TypeError` these exceptions signal that, for inference brains, the usual continuation is to raise `UseInferenceDefault` from that as a signal that that particular brain should be ignored. Here I'm trying to follow the same convention, if some caller calls `getitem` with a function that doesn't match the required signature then we know that Python would raise a `TypeError` at this point and so that there's no reasonable inference to return from this function. We could raise `InferenceError` I suppose, I was just trying to be *more* specific.

Reply:
e.g. see `ClassDef.getitem` for an existing example: https://github.com/PyCQA/astroid/blob/d66e3d113c0b2cb9283f3518e5a494d139c6779a/astroid/scoped_nodes.py#L2661-L2674

Reply:
It definitely makes sense. Thanks for the explanation!